clients/unshaded/src/main/java/tachyon/client/file/TachyonFileSystemUtils.java

```diff
@@ -134,4 +142,42 @@ public static boolean waitCompleted(final TachyonFileSystemCore tfs, final Tachy
 
     return completed;
   }
+
+  /**
+   * Persist the given file to the under file system.
+   *
+   * @param tfs {@link TachyonFileSystem} to carry out tachyon operations
+   * @param file the file to persist
+   * @param fileInfo the file info of the file
+   * @param tachyonConf TachyonConf object
+   * @return the size of the file persisted
+   * @throws IOException if an I/O error occurs
+   * @throws FileDoesNotExistException if the given file does not exist
+   * @throws TachyonException if an unexpected Tachyon error occurs
+   */
+  public static long persistFile(TachyonFileSystem tfs, TachyonFile file, FileInfo fileInfo,
```

Comment:
Yeah I put that in because the shell needed to call ``getFileInfo`` earlier to determine whether we were persisting a directory or a file, so I didn't want to have to make a redundant RPC call.

I guess currently the TachyonConf is whatever the shell passes it, which seems like it's just a default TachyonConf (so not from ``ClientContext``), but since client methods can now call this, it could probably come from a ``ClientContext`` too.